### DISCUSSION TITLE
Signal level voltage on DSC PC5020 Rev A

### DISCUSSION THREAD

**Discussion by gunku** [taligentx/dscKeybusInterface#249] [dscKeybusInterface Discussion]
Hi!

Just built a board based on NodeMCU 1.0 and trying to run WebKeypad script on it. However I've noticed that if I am not connecting CLK and DATA wires, I can access WEB interface through WiFi, but as soon as I attach them WiFi response stops.

Anyway to debug I've tried to eliminate potential problems and since I had 30k resistors instead of 33k in voltage divier, decided to check with osciloscope what is a signal voltage on DSC panel. To my surprise it wasn't ~12V as mentioned on this site, but around 1.4V. That would mean that I could run signal directly into NodeMCU without divider.

I wonder if anyone else has seen this as I cannot find exact signal voltage levels for PC5020 in manuals?

Thanks for every contributor for a great peace of software!

With best regards,
Gundars (From Latvia)
![IMG_7417](https://user-images.githubusercontent.com/5734424/124377756-4c77a480-dcb6-11eb-9d9d-28ceec4ececd.jpeg)

**Comment by gunku** [comment]
Got an answer to voltage mismatches, but not to why it doesn't work :) On Voltages it was a simple stupid human error (mine) - I've set a probe to 10x without adjusting oscilloscope settings, so voltage actually is 14.8V peak to peak. Good thing is that I've found it out without running signal straight into NodeMCU :D

**Comment by Dilbert66** [comment]
Are you using ESP library 3.0.0? If so there is a bug in isr handling that will cause it to crash.   The fact that it does not crash when the dsc lines are not connecting is because the chip is not executing ISR routines since there are no interrupts generated. Try updating to 3.0.1.

**Reply by gunku** [reply]
Thank you for very valuable input!

I was on 3.0.0. but upgraded to recently released 3.0.1. Now I get lots of info on serial monitor, when reading Keybus reader script and Web keyboard doesn't crash as soon as I attach data signal wires, but rather when I try to press a key in web interface.

Crash logs tsrts with:

User exception (panic/abort/assert)
--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Panic core_esp8266_main.cpp:137 __yield

>>>stack>>>

**Comment by Dilbert66** [comment]
Which branch are you using? master or develop?

**Reply by gunku** [reply]
Master

**Comment by Dilbert66** [comment]
Try the develop branch. It includes a few fixes and updates. It's pretty stable.
